lib/class-wp-rest-menus-controller.php

```diff
@@ -345,6 +422,40 @@ public function delete_item( $request ) {
 		return $response;
 	}
 
+	/**
+	 * Updates the menu's locations from a REST request.
+	 *
+	 * @param int             $menu_id The menu id to update the location form.
+	 * @param WP_REST_Request $request The request object with menu and locations data.
+	 *
+	 * @return true|WP_Error WP_Error on an error assigning any of the locations, otherwise null.
+	 */
+	protected function handle_locations( $menu_id, $request ) {
+		if ( ! isset( $request['locations'] ) ) {
+			return true;
+		}
+
+		$menu_locations = get_registered_nav_menus();
+		$menu_locations = array_keys( $menu_locations );
+		$new_locations  = array();
+		foreach ( $request['locations'] as $location ) {
+			if ( ! in_array( $location, $menu_locations, true ) ) {
+				return new WP_Error( 'check_assign_locations_permission', __( 'Menu location does not exist.' ), array( 'status' => 400 ) );
```

Comment:
```suggestion
				return new WP_Error( 'invalid_menu_location', __( 'Menu location does not exist.' ), array( 'status' => 400 ) );
```

This isn't actually a permissions check.